src/widgets/pathsViewer.js

```diff
@@ -132,8 +125,8 @@ var FlatsealPathsViewer = GObject.registerClass({
         row.text = path;
         row.connect('remove-requested', this._remove.bind(this, row));
         row.connect('notify::text', this._changed.bind(this));
-        row.show();
 
-        this._box.pack_end(row, true, true, 0);
+        this.append(row);
```

Comment:
```suggestion
        this.prepend(row);
```

Fixes the issue:
> permissions: "Overridden by" badges are positioned on the wrong row, e.g. go to Flatseal, "other files" add a new row "/newpath", then you will see:

---

Changing from GTK3's GtkBox.pack_end to GTK4's GtkBox.append resulted in the ordering of the child rows being reversed. The existing logic to set the status icon for the rows is reliant on said ordering, hence the incorrectly set icons (on opening an app's page with existing overrides). This is the easiest fix and maintains the existing UI :slightly_smiling_face:

Reply:
The problem was about the badge, not the row. In any case its kinda weird UI to prepend rows, ill fix it to match previous behaviour

Reply:
https://github.com/A6GibKm/Flatseal/blob/gtk4-port/src/widgets/pathsViewer.js#L90 Yes, I'm aware, the logic for setting the badge was dependent on the ordering